File: samples/JavaScript/closest-pair-problem.js
function distance(p1, p2) {
  var dx = Math.abs(p1.x - p2.x);
  var dy = Math.abs(p1.y - p2.y);
  return Math.sqrt(dx*dx + dy*dy);
}

function bruteforceClosestPair(arr) {
  if (arr.length < 2) {
    return Infinity;
  } else {
    var minDist = distance(arr[0], arr[1]);
    var minPoints = arr.slice(0, 2);

    for (var i=0; i<arr.length-1; i++) {
      for (var j=i+1; j<arr.length; j++) {
        if (distance(arr[i], arr[j]) < minDist) {
          minDist = distance(arr[i], arr[j]);
          minPoints = [ arr[i], arr[j] ];
        }
      }
    }
    return {
      distance: minDist,
      points: minPoints
    };
  }
}

File: samples/Scala/closest-pair-problem.scala
import scala.util._

object ClosestPair{
   class Point(x:Double, y:Double) extends Pair(x,y){
      def distance(p:Point)=math.hypot(_1-p._1, _2-p._2)
   }

   def closestPairBF(a:Array[Point])={
      var minDist=a(0) distance a(1)
      var minPoints=(a(0), a(1))

      for(p1<-a; p2<-a if p1.ne(p2); dist=p1 distance p2 if(dist<minDist)){
         minDist=dist;
         minPoints=(p1, p2)
      }
      (minPoints, minDist)
   }

   def main(args: Array[String]): Unit = {
      val a=Array.fill(1000)(new Point(Random.nextDouble, Random.nextDouble))
      val (points, dist)=closestPairBF(a)
      println("min: "+points._1+" - "+points._2+"   ->"+dist)
   }
}
